### PR TITLE
Payara Micro application deployment testcase

### PR DESCRIPTION
**Pull request by jGauravGupta**
## Description
This is a test case to deploy sample web application to Payara Micro instance locally(similar to David's docker container tests).

### CODE REVIEW

**Comment by jGauravGupta** [comment]
jenkins test please

File: appserver/tests/payara-samples/micro-programmatic/src/test/java/fish/payara/samples/programatic/DeployApplicationTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    private static final File WAR_FILE = createWar();
+    private static final String WEBAPP_CONTEXT = "/" + WAR_FILE.getName().substring(0, WAR_FILE.getName().length() - 4);
+    private static final String HOST_NAME = System.getProperty("payara.adminHost", "localhost");
+
+    @Test
+    public void deployApplicationInVanillaMode() throws Exception {
+        PayaraMicroServer server = PayaraMicroServer.newInstance();
+        server.start("--autobindhttp", "--nocluster", "--deploy", WAR_FILE.getAbsolutePath());
+        assertEquals(TestServlet.RESPONSE_TEXT, download(HOST_NAME, server.getHttpPort()));
+        server.stop();
```
Comment (dmatej) [inline appserver/tests/payara-samples/micro-programmatic/src/test/java/fish/payara/samples/programatic/DeployApplicationTest.java:64]: I would move the server variable to a field and the server.stop to a new method:
```
@After
public void stopServer() {
  if (server != null) {
    server.stop();
  }
}
```

**Review by dmatej — CHANGES_REQUESTED** [review]

**Comment by jGauravGupta** [comment]
jenkins test please

**Review by dmatej — APPROVED** [review]
Passed on my machine :-)

`mvn clean install -Pfastest,QuickBuild -TC6; mvn clean install -Ppayara-samples,payara-micro-managed`

**Pull request merged**